ui/components/app/clear-metametrics-data/clear-metametrics-data.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useContext } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  hideDeleteMetaMetricsDataModal,
+  markingMetaMetricsDataDeletion,
+  openDataDeletionErrorModal,
+} from '../../../ducks/app/app';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  Modal,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '../../component-library';
+import {
+  AlignItems,
+  BlockSize,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import {
+  createMetaMetricsDataDeletionTask,
+  setHasMetaMetricsDataRecorded,
+} from '../../../store/actions';
+import { getParticipateInMetaMetrics } from '../../../selectors';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+
+export default function ClearMetaMetricsData() {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+
+  const participateInMetaMetrics = useSelector(getParticipateInMetaMetrics);
+
+  const closeModal = () => {
+    dispatch(hideDeleteMetaMetricsDataModal());
+  };
+
+  const deleteMetaMetricsData = async () => {
+    try {
+      await createMetaMetricsDataDeletionTask();
+      dispatch(markingMetaMetricsDataDeletion());
+      if (participateInMetaMetrics) {
+        await setHasMetaMetricsDataRecorded(true);
+        trackEvent({
+          category: MetaMetricsEventCategory.Settings,
+          event: MetaMetricsEventName.MetricsDataDeletionRequest,
+        });
+      } else {
+        await setHasMetaMetricsDataRecorded(false);
+      }
+    } catch (error: unknown) {
```

Comment:
Should we also track an error event here? e.g. `MetaMetricsEventName.MetricsDataDeletionRequestError`

cc @gauthierpetetin

Reply:
Hi @danjm ,

In Mobile, we've started to [track expected errors as Analytics](https://github.com/MetaMask/metamask-mobile/blob/main/app/util/metrics/TrackError/trackErrorAsAnalytics.ts) in order to reduce the noise in Sentry, which shall be kept for unexpected errors.
To achieve that we've introduced a generic `Error occured` event that we want to use for all our errors.

Do we have an equivalent generic event for errors in Extension?
- If yes we could use it
- If not, it might be a good occasion to introduce such an event in Extension as well, wdyt?

cc @worldlyjohn @Cal-L for visibility

Reply:
@NiranjanaBinoy Can you add an 'Error Occured' metrics event, and then track that event here?